packages/block-library/src/post-terms/index.php

```diff
@@ -35,21 +32,16 @@ function render_block_core_post_terms( $attributes, $content, $block ) {
 		$classes .= ' has-text-align-' . $attributes['textAlign'];
 	}
 
-	$terms_links = '';
-	foreach ( $post_terms as $term ) {
-		$terms_links .= sprintf(
-			'<a href="%1$s">%2$s</a> | ',
-			get_term_link( $term->term_id ),
-			esc_html( $term->name )
-		);
-	}
-	$terms_links        = trim( $terms_links, ' | ' );
+	$separator = empty( $attributes['separator'] ) ? ' ' : $attributes['separator'];
+
 	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $classes ) );
 
-	return sprintf(
-		'<div %1$s>%2$s</div>',
-		$wrapper_attributes,
-		$terms_links
+	return get_the_term_list(
+		$block->context['postId'],
+		$attributes['term'],
+		"<div $wrapper_attributes>",
+		'<span class="wp-block-post-terms__sep">' . $separator . '</span>',
```

Comment:
Should we add this if no `separator` is provided?

Reply:
If no separator is provided then it falls back to an empty space, and the theme may actually be theming this with something like
```css
.wp-block-post-terms__sep:before { /* theme styles */ }
```